tmt/base.py

```diff
@@ -898,9 +928,11 @@ def _lint_discover(self):
     def _lint_discover_fmf(discover):
         """ Lint fmf discover method """
         # Validate remote id and translate to human readable errors
-        valid, error = tmt.utils.validate_fmf_id(
-            {key: value for key, value in discover.items()
-                if key in ['url', 'ref', 'path']})
+        valid, error = FmfId(**{
```

Comment:
I'd suggest splitting the data collection and validation to 2 lines, it IMHO makes splitting less awkward and hence the code more readable. Something like

```python
        fmf_id_data = {key: value for key, value in discover.items()
                       if key in ['url', 'ref', 'path']}
        valid, error = FmfId(**fmf_id_data).validate()
```